kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,605 @@
+    replicaCount=$((replicaCount+1))
+  elif [ "${withReplicas}" == "DECREASED" ]; then
+    replicaCount=$((replicaCount-1))
+  fi
+  for localServerName in ${sortedByAlwaysServers[@]:-}; do
+    getEffectivePolicy "${domainJson}" "${localServerName}" "${clusterName}" policy
+    # Update policy when server name matches current server and unsetting
+    if [[ "${serverName}" == "${localServerName}" && "${withPolicy}" == "UNSET" ]]; then
+      policy=UNSET
+    fi
+    # check if server should start based on replica count, policy and current replicas
+    shouldStart "${currentReplicas}" "${policy}" "${replicaCount}" result
+    if [ "${result}" == 'true' ]; then
+      # server should start, increment current replicas and add server to list of started servers
+      currentReplicas=$((currentReplicas+1))
+      startedServers+=(${localServerName})
+    fi
+  done
+  startedSize=${#startedServers[@]}
+  if [ ${startedSize} -gt 0 ]; then
+    # check if current server is in the list of started servers
+    if checkStringInArray ${serverName} ${startedServers[@]}; then
+      eval $__started="true"
+      return
+    fi
+  fi
+  eval $__started="false"
+}
+
+#
+# Function to check if server should start based on policy and current replicas
+# $1 - Current number of replicas
+# $2 - Server start policy
+# $3 - Replica count
+# $4 - Returns "true" or "false" indicating if server should start.
+#
+function shouldStart {
+  local currentReplicas=$1
+  local policy=$2
+  local replicaCount=$3 
+  local __result=$4
+
+  if [ "$policy" == "ALWAYS" ]; then
+    eval $__result=true
+  elif [ "$policy" == "NEVER" ]; then
+    eval $__result=false
+  elif [ "${currentReplicas}" -lt "${replicaCount}" ]; then
+    eval $__result=true
+  else 
+    eval $__result=false
+  fi
+}
+
+#
+# Function to create patch string for updating replica count
+# $1 - Domain resource in json format
+# $2 - Name of cluster whose replica count will be patched
+# $3 - operation string indicating whether to increment or decrement replica count. 
+#      Valid values are "INCREMENT" and "DECREMENT"
+# $4 - Return value containing replica update patch string
+# $5 - Return value containing updated replica count
+#
+function createReplicaPatch {
+  local domainJson=$1
+  local clusterName=$2
+  local operation=$3
+  local __result=$4
+  local __replicaCount=$5
+  local maxReplicas=""
+  local infoMessage="Current replica count value is same as maximum number of replica count. \
+Not increasing replica count value."
+
+  maxReplicaCmd="(.status.clusters[] | select (.clusterName == \"${clusterName}\")) \
+    | .maximumReplicas"
+  getReplicaCount  "${domainJson}" "${clusterName}" replica
+  if [ "${operation}" == "DECREMENT" ]; then
+    replica=$((replica-1))
+    if [ ${replica} -lt 0 ]; then
+      replica=0
+    fi
+  elif [ "${operation}" == "INCREMENT" ]; then
+    maxReplicas=$(echo ${domainJson} | jq "${maxReplicaCmd}")
+    if [ ${replica} -eq ${maxReplicas} ]; then
```

Comment:
Just in case the replica count already happens to be above max, maybe change to -ge and modify the info message accordingly.